### DISCUSSION TITLE
CA for code signing

### DISCUSSION THREAD

**Discussion by MrPnut** [community/community#19709] [Other Feature Feedback, Questions, & Ideas]
Options for code signing as an open source project maintainer is pretty limited.  Code signing certs are 80-100 USD / year for cheap ones that don't necessarily even clear smartscreen filters.

My "request" is to find out if it's possible for Github to start its own CA for projects it hosts?  Basically where projects "in good standing", or popular / verified open source projects can get a code signing cert.  

We have letsencrypt for web but nothing for native applications.

**Comment by zapek** [comment]
I second that.

GitHub uses 2FA and people using GitHub sponsors even have gone through a validation similar to what CAs do.

Besides, the whole source and commit log is available to inspect the resulting build. You can hardly be more transparent than that. It could also be restricted to GitHub Actions to avoid having the certificate "in the wild" if that is the problem.
